compiler/rustc_trait_selection/src/traits/query/evaluate_obligation.rs

```diff
@@ -67,21 +68,22 @@ impl<'tcx> InferCtxt<'tcx> {
 
         if self.next_trait_solver() {
             self.probe(|snapshot| {
-                let mut fulfill_cx = crate::solve::FulfillmentCtxt::new(self);
-                fulfill_cx.register_predicate_obligation(self, obligation.clone());
-                // True errors
-                // FIXME(-Znext-solver): Overflows are reported as ambig here, is that OK?
```

Comment:
either handle this FIXME or keep it :<

Reply:
I think this is fine to remove. We don't have `EvaluatedToErrStackDependent` anymore, and `EvaluatedToAmbigStackDependent` only matters for the purposes of caching in the solver.